### PR TITLE
Implement Campaign Management Functions

### LINKED ISSUE #99: Implement Campaign Management Functions in ICampaignDonation
The `ICampaignDonation` interface defines three critical campaign management functions that need to be implemented to provide campaign owners with proper control over their fundraising campaigns.

## Code Reference
```cairo:167-204:src/interfaces/ICampaignDonation.cairo
// *************************************************************************
//                        CAMPAIGN MANAGEMENT
// *************************************************************************

/// Updates the target amount for a campaign (only by owner before any donations)
///
/// # Arguments
/// * `campaign_id` - The campaign ID
/// * `new_target` - The new target amount
///
/// # Requirements
/// * Caller must be campaign owner
/// * Campaign must have zero balance
/// * New target must be greater than zero
fn update_campaign_target(ref self: TContractState, campaign_id: u256, new_target: u256);

/// Cancels a campaign and enables refunds (only if no withdrawals have occurred)
///
/// # Arguments
/// * `campaign_id` - The campaign ID
///
/// # Requirements
/// * Caller must be campaign owner
/// * Campaign must not be withdrawn
/// * Campaign must not have reached its goal
fn cancel_campaign(ref self: TContractState, campaign_id: u256);

/// Allows donors to claim refunds from cancelled campaigns
///
/// # Arguments
/// * `campaign_id` - The campaign ID
///
/// # Requirements
/// * Campaign must be cancelled
/// * Caller must have donated to the campaign
/// * Refund must not have been claimed already
fn claim_refund(ref self: TContractState, campaign_id: u256);
```

## Tasks
- [ ] Implement `update_campaign_target` function
  - [ ] Verify caller is campaign owner
  - [ ] Check campaign balance is zero
  - [ ] Validate new target amount > 0
  - [ ] Update campaign target in storage
  - [ ] Emit appropriate event

- [ ] Implement `cancel_campaign` function
  - [ ] Verify caller is campaign owner
  - [ ] Check campaign hasn't been withdrawn
  - [ ] Check campaign hasn't reached its goal
  - [ ] Update campaign status to cancelled
  - [ ] Emit cancellation event

- [ ] Implement `claim_refund` function
  - [ ] Verify campaign is cancelled
  - [ ] Check caller has donated to the campaign
  - [ ] Ensure refund hasn't been claimed already
  - [ ] Calculate refund amount
  - [ ] Transfer tokens back to donor
  - [ ] Mark refund as claimed
  - [ ] Emit refund event

## Technical Considerations
1. **State Management**: Need to add a `cancelled` field to the `Campaigns` struct
2. **Refund Tracking**: Implement a mapping to track claimed refunds per donor per campaign
3. **Events**: Define and emit events for:
   - `CampaignTargetUpdated`
   - `CampaignCancelled`
   - `RefundClaimed`
4. **Security**: Ensure proper access control and prevent reentrancy attacks

## Acceptance Criteria
- [ ] All three functions are fully implemented with proper validation
- [ ] Comprehensive unit tests cover all scenarios including edge cases
- [ ] Gas optimization is considered for refund claims
- [ ] Events are properly emitted for all state changes
- [ ] Documentation is updated with implementation details

## Priority
Medium - These functions are essential for proper campaign lifecycle management and user trust

### PR DESCRIPTION
**Pull request by wheval**
Closes  #99 

## Summary by CodeRabbit

- **New Features**
  - Added ability for campaign owners to update the campaign target before any donations are made.
  - Introduced campaign cancellation, allowing owners to cancel campaigns under specific conditions.
  - Enabled donors to claim refunds for cancelled campaigns.
  - Added new events to track campaign updates, cancellations, and refunds.
  - Expanded error messages to cover more specific campaign and donation-related conditions.

- **Documentation**
  - Improved and reformatted documentation for campaign management functions in the interface.

- **Tests**
  - Added comprehensive tests covering campaign target updates, cancellations, and refund claims including success and failure scenarios.

### CODE REVIEW

File: src/campaign_donation.cairo [hunk 1]
```diff
@@ -296,6 +336,93 @@ pub mod CampaignDonation {
...
             campaign
         }
+
+        fn update_campaign_target(ref self: ContractState, campaign_id: u256, new_target: u256) {
+            let caller = get_caller_address();
+            assert(campaign_id > 0, CAMPAIGN_NOT_FOUND);
+            assert(campaign_id <= self.campaign_counts.read(), CAMPAIGN_NOT_FOUND);
+            let mut campaign = self.get_campaign(campaign_id);
+            assert(caller == campaign.owner, CALLER_NOT_CAMPAIGN_OWNER);
+            assert(campaign.current_balance == 0, CAMPAIGN_HAS_DONATIONS);
```
Comment (mubarak23) [inline src/campaign_donation.cairo:346]: so you can only update campaign target if the campaign does not have any donation

**Comment by mubarak23** [comment]
@wheval plese, let push commit the right way, alway run format on the code, 

the CI run three thing,
- fomat code
- build 
- run test
- 

if anyone of this fail, the PR will not get merged, 

all the CI must pass 

run them locally before pushing your code



the moment the CI is Failing because the code is not formatted , i expect a fix ASAP
